python/tests/test_fast_sdpa.py

```diff
@@ -32,9 +32,82 @@ def repeat(a):
     return mlx_primitives_sdpa(q, k, v, scale)
 
 
-class TestFastSDPA(mlx_tests.MLXTestCase):
+class TestFastSelfAttentionSDPA(mlx_tests.MLXTestCase):
     def test_fast_sdpa(self):
 
+        # Not yet supported:
+        # * K pre-transposed in kernel, V pre-transposed in kernel
+        np.random.seed(0)
+        R = 20
+        L = R
+        Dk = 64
+        H = 3
+        scale = float(1.0 / np.sqrt(Dk))
+        q_npy = np.random.normal(0.0, 1.0, (1, H, R, Dk)).astype(np.float32)
+        k_npy = np.random.normal(0.0, 1.0, (1, H, L, Dk)).astype(np.float32)
+        v_npy = np.random.normal(0.0, 1.0, (1, H, L, Dk)).astype(np.float32)
+
+        q_mlx = mx.array(q_npy)
+        k_mlx = mx.array(k_npy)
+        v_mlx = mx.array(v_npy)
+
+        reference = mlx_primitives_sdpa(q_mlx, k_mlx, v_mlx, scale)
+
+        o_mlx = mx.fast.scaled_dot_product_attention(
+            q_mlx, k_mlx, v_mlx, scale=scale, mask=None
+        )
+
+        self.assertListEqual(list(reference.shape), list(o_mlx.shape))
+        self.assertTrue(mx.allclose(o_mlx, reference, atol=1e-4))
+
+        dtypes = [np.float32]
+
+        Dk = 64
+
+        if self.is_apple_silicon:
+            dtypes.append(np.half)
+
+        for SEQUENCE_LENGTH in [63, 129, 400, 1178, 2202]:
+            for DTYPE in dtypes:
```

Comment:
We might want to make these test cases a bit smaller and then move a more comprehensive set to our longer running test suite (which is not public). Our CI test machines are not that big..